### PR TITLE
feat: [CDS-54917]: combined expression for usergroups

### PR DESCRIPTION
**Pull request by naman-goenka**
# Harness Developer Pull Request
Thanks for helping us make the Developer Hub better. The PR will be looked at
by the maintainers. 

## What Type of PR is This?

- [ ] Issue
- [ ] Feature
- [ ] Maintenance/Chore

If tied to an Issue, list the Issue(s) here:

* *Issue(s)*

## House Keeping
Some items to keep track of. Screen shots of changes are optional but would help the maintainers review quicker. 

- [ ] Tested Locally
- [ ] *Optional* Screen Shoot.

### CODE REVIEW

File: docs/platform/9_Approvals/adding-harness-approval-stages.md [hunk 1]
```diff
@@ -124,6 +124,29 @@ For example, in a subsequent step's **Conditional Execution** settings, you coul
...
+
+For example, combined expression for usergroups field can be 
+
+`<+ <+pipeline.variables.grouplist1> + "," + <+pipeline.variables.grouplist2> >.split(",")`
+
+where the expression `<+pipeline.variables.grouplist1>` resolves to the value : `_project_all_users,testUserGroup`
+and the expression `<+pipeline.variables.grouplist2>` resolves to the value : `org._organization_all_users,account._account_all_users`. The above two expressions resolve to a string of comma separated user groups. The combined expression will resolve to a list of 4 user groups (`_project_all_users`, `testUserGroup`, `org._organization_all_users` and `account._account_all_users`). 
+
+     :::important
+     Please ensure no spaces are given in expression values for example `_project_all_users,testUserGroup` to match the split criteria exactly which is `,` in this example.
```
Comment (rashminandasahoo) [inline docs/platform/9_Approvals/adding-harness-approval-stages.md:145]:
```suggestion
     Make sure no spaces are present in the expression values. In the above example, the format must be `_project_all_users,testUserGroup` to match the split criteria `,` . 
```

**Review by rashminandasahoo — CHANGES_REQUESTED** [review]

File: docs/platform/9_Approvals/adding-harness-approval-stages.md [hunk 2]
```diff
@@ -124,6 +124,29 @@ For example, in a subsequent step's **Conditional Execution** settings, you coul
 
 `<+pipeline.stages.Shell_Script.spec.execution.steps.Harness_Approval_Step.output.approverInputs.foo> == 1`
 
+### Option: User Groups as Expressions
+
+Select Expression tab for **User Groups** field. User Groups in Harness Approvals can be provided as expressions in two ways:
```
[the remark this answers is not in the document]
Comment (naman-goenka) [reply docs/platform/9_Approvals/adding-harness-approval-stages.md:129]: done

File: docs/platform/9_Approvals/adding-harness-approval-stages.md [hunk 3]
```diff
@@ -124,6 +124,29 @@ For example, in a subsequent step's **Conditional Execution** settings, you coul
...
+
+* **Individual Expression**: Select **Individual** option to specify individual expressions for each user group that needs to be configured as Approvers. Add more user groups by clicking on **Add** button.  
+
+![](./static/adding-harness-approval-stages-18.png)
+
+* **Combined Expression**: Select **Combined** option to specify a combined expression for all user groups that needs to be configured as Approvers. This expression should resolve in a list of strings.
```
[the remark this answers is not in the document]
Comment (naman-goenka) [reply docs/platform/9_Approvals/adding-harness-approval-stages.md:135]: done

File: docs/platform/9_Approvals/adding-harness-approval-stages.md [hunk 4]
```diff
@@ -124,6 +124,29 @@ For example, in a subsequent step's **Conditional Execution** settings, you coul
...
+
+For example, combined expression for usergroups field can be
```
[the remark this answers is not in the document]
Comment (naman-goenka) [reply docs/platform/9_Approvals/adding-harness-approval-stages.md:137]: done

File: docs/platform/9_Approvals/adding-harness-approval-stages.md [hunk 5]
```diff
@@ -124,6 +124,29 @@ For example, in a subsequent step's **Conditional Execution** settings, you coul
...
+
+![](./static/adding-harness-approval-stages-18.png)
+
+* **Combined Expression**: Select **Combined** option to specify a combined expression for all user groups that needs to be configured as Approvers. This expression should resolve in a list of strings.
+
+For example, combined expression for usergroups field can be 
+
+`<+ <+pipeline.variables.grouplist1> + "," + <+pipeline.variables.grouplist2> >.split(",")`
+
+where the expression `<+pipeline.variables.grouplist1>` resolves to the value : `_project_all_users,testUserGroup`
```
[the remark this answers is not in the document]
Comment (naman-goenka) [reply docs/platform/9_Approvals/adding-harness-approval-stages.md:141]: done

File: docs/platform/9_Approvals/adding-harness-approval-stages.md (hunk 1 above)
Comment (naman-goenka) [reply docs/platform/9_Approvals/adding-harness-approval-stages.md:145]: done

File: docs/platform/9_Approvals/adding-harness-approval-stages.md [hunk 6]
```diff
@@ -124,6 +124,29 @@ For example, in a subsequent step's **Conditional Execution** settings, you coul
...
+and the expression `<+pipeline.variables.grouplist2>` resolves to the value : `org._organization_all_users,account._account_all_users`. The above two expressions resolve to a string of comma separated user groups. The combined expression will resolve to a list of 4 user groups (`_project_all_users`, `testUserGroup`, `org._organization_all_users` and `account._account_all_users`).
```
[the remark this answers is not in the document]
Comment (naman-goenka) [reply docs/platform/9_Approvals/adding-harness-approval-stages.md:142]: done

**Review by rashminandasahoo — APPROVED** [review]

**Pull request merged**